docs/components/renderless/kstate.md

```diff
@@ -0,0 +1,78 @@
+# KState
```

Comment:
Does this PR require KState? Can the segment group use parent component state? Little confused on the need to contain state in a wrapper PR when it would most likely be in the original component/page to begin with.

Reply:
It makes the segmented control docs a lot less noisy

Reply:
ideally these would be separate PR's. This PR is getting a little big, especially since KState has some review comments and it exists only right now to improve the doc experience.

Reply:
Yeah, 1 thing we can do is turn KState into a docs only component to help manage the state in md files. IDK that we have a real need in applications for it at the moment. Open to other suggestions though

Reply:
I agree, happy to pull KState into a separate PR this one depends on, if it helps.